File: trace_viewer/tracing/importer/importer.js
// Copyright (c) 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

'use strict';

/**
 * @fileoverview Base class for trace data importers.
 */

tvcm.exportTo('tracing.importer', function() {

  function Importer() {
  }

  Importer.prototype = {
    __proto__: Object.prototype,

    /**
     * Called by the Model to extract one or more subtraces from the event data.
     */
    extractSubtraces: function() {
      return [];
    },

    /**
     * Called to import events into the Model.
     */
    importEvents: function() {
    },

    /**
     * Called to import sample data into the Model.
     */
    importSampleData: function() {
    },

    /**
     * Called by the Model after all other importers have imported their
     * events.
     */
    finalizeImport: function() {
    },

    /**
     * Called by the Model to join references between objects, after final
     * model bounds have been computed.
     */
    joinRefs: function() {
    }
  };

  return {
    Importer: Importer
  };
});

File: telemetry/telemetry/timeline/importer.py
# Copyright 2014 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

class TimelineImporter(object):
  """Interface for classes that can add events to
  a timeline model from an TimelineData."""
  def __init__(self, model, timeline_data, import_priority=0):
    self._model = model
    self._timeline_data = timeline_data
    self.import_priority = import_priority

  @staticmethod
  def CanImport(event_data_wrapper):
    """Returns true if the importer can process the given event data in the
    wrapper."""
    raise NotImplementedError

  def ImportEvents(self):
    """Processes the event data in the wrapper and creates and adds
    new timeline events to the model"""
    raise NotImplementedError

  def FinalizeImport(self):
    """Called after all other importers for the model are run."""
    raise NotImplementedError
